docs/format_howtos.py

```diff
@@ -0,0 +1,79 @@
+"""
+Read all of the HOWTO .diff files and convert them into .html files
+that are both Python syntax highlighted /and/ diff syntax highlighted.
+
+Then these can be included directly in readthedocs as inline HTML
+files.
+"""
+
+import pygments
+import pygments.formatters
+from pygments.lexers import PythonLexer
+
+import os
+
+def main():
+  os.makedirs('_formatted_howtos', exist_ok=True)
+  for diff_filename in os.listdir(os.path.join('..', 'howtos', 'diffs')):
+    format_howto(os.path.join('..', 'howtos', 'diffs', diff_filename),
+           os.path.join('_formatted_howtos', diff_filename + '.html'))
+
+def format_howto(input_file, output_file):
+  # Load one of our HOWTO diff files.
+  with open(input_file) as f:
+    diff = f.readlines()
+
+  # Ignore the first two line, which look like:
+  #
+  # diff --git a/examples/mnist/train.py b/examples/mnist/train.py
+  # index 51d2fde..a9d7dcb 100644
+  diff = diff[2:]
+
+  # Remove more than one empty line in a row
+  diff = [diff[lineno] for lineno in range(len(diff))
```

Comment:
Are you trying to remove double newlines? It is not entirely clear to me from the comment. Perhaps it is easier to understand if you use a regex? https://stackoverflow.com/a/28902081

Reply:
Well the file is already split into lines, so you can't use a regexp. (And the lines aren't actually empty, they contain the first space that designates whether it's an addition or removal of a line, or just context).

I'll modify the comment.

Reply:
I pushed another change rewriting the logic a bit and adding a clarifying comment. Please take another look.

Reply:
Thanks for the clarifying comment!